src/extrema_set.rs

```diff
@@ -0,0 +1,42 @@
+/// Implementation guts for `min_set`, `min_set_by`, and `min_set_by_key`.
+pub fn min_set_impl<I, K, F, L>(mut it: I,
+                                mut key_for: F,
+                                mut lt: L) -> Option<Vec<I::Item>>
```

Comment:
What do you think about returning an empty `Vec` instead of `None`? I think it would make things a bit more clear, especially since the types do not indicate that a `Some` always contains a non-empty vector.

Reply:
Do you think we could/should be a bit more generic than `Vec` here? I.e. should we try to be like `collect` (which returns something `FromIterator<Self::Item>` and lets the caller choose)?

Reply:
I personally think that returning an Option is a good choice. In application code that I write, when using max or min functionality of sequences, an empty sequence is often an indication that something has gone awry and should be forced to be handled in another way. I imagine that in most cases, the Option-ness can be handled by the `?`-operator. Also, it matches the signature for max/min from the standard library, which returns an `Option` of an `Item`. I can naturally change it, but I do think that Option makes sense here.

As for why Vec and not some kind of iterator. The number of results is unknown before computing the result, and the result is unknown before going through the whole input iterator, which implies that some kind of temporary storage area is needed. Since the data is already in a Vec, I thought that the least convoluted thing was to just give the user back that. We can of course wrap the already constructed `Vec` in something that exposes it like a `FromIterator`, I have no real opinion here.

Reply:
Regarding the `Vec`: I naively thought about substituting the whole `Vec` even within the implementation (basically allowing to customize the kind of temporary storage), so that no conversion would be required upon returning it.

However, I see that `FromIterator` alone is possibly a bit too narrow to acchieve that. Maybe any container satisfying `Default + Extend` or `FromIterator + Extend` could be used?

Reply:
Since the code by neccesity (unless double iteration is used) _provisionally_ accumulates an unbounded number of result items, and discards these when a new provisionally best results is found, something more than `FromIterator + Extend` would be needed AFAIK. 

My guess would be that anything that also has the possibility to clear the results is no better than the interal `Vec`, and would also be less clear. I'd be glad to learn I'm wrong though!